src/Simplify.elm

```diff
@@ -1352,10 +1353,64 @@ expressionVisitorHelp node context =
                 _ ->
                     onlyErrors []
 
+        Expression.RecordAccess record field ->
+            let
+                fieldName =
+                    Node.value field
+            in
+            case Node.value record of
+                Expression.RecordExpr setters ->
+                    onlyErrors (recordAccessChecks node fieldName setters)
+
+                Expression.RecordUpdateExpression _ setters ->
+                    onlyErrors (recordAccessChecks node fieldName setters)
+
+                _ ->
+                    onlyErrors []
+
         _ ->
             onlyErrors []
 
 
+recordAccessChecks : Node Expression -> String -> List (Node RecordSetter) -> List (Error {})
+recordAccessChecks node fieldName setters =
+    let
+        setterValues =
+            List.map
+                (\setter ->
+                    let
+                        ( nfield, nvalue ) =
+                            Node.value setter
+                    in
+                    ( Node.value nfield, nvalue )
+                )
+                setters
+
+        nodeRange =
+            Node.range node
+    in
+    case
+        List.Extra.find (\( setterField, _ ) -> setterField == fieldName) setterValues
```

Comment:
There is already a `find` function defined in the same module. No need for the dependency.

Reply:
Ok, let me rewrite it